lib/collection/src/shards/replica_set/mod.rs

```diff
@@ -789,6 +795,39 @@ pub enum ReplicaState {
     PartialSnapshot,
 }
 
+#[derive(Copy, Clone, Debug)]
+struct SyncState {
+    last_attempt: Instant,
+    backoff: Duration,
+}
+
+impl Default for SyncState {
+    fn default() -> Self {
+        Self {
+            last_attempt: Instant::now(),
+            backoff: Duration::from_secs(0),
+        }
+    }
+}
+
+impl SyncState {
+    pub fn retry(&mut self) -> bool {
```

Comment:
Nitpick, but maybe we should renamed this to something like `should_retry_now()`.

Reply:
Done.